File: backend/app/schemas/metrics.py
from datetime import date
from typing import Optional

from pydantic import BaseModel, ConfigDict


class TodayMetrics(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    date: date
    readiness_score: Optional[int] = None
    readiness_level: Optional[str] = None
    hrv_last_night_avg: Optional[int] = None
    hrv_status: Optional[str] = None
    sleep_seconds: Optional[int] = None
    avg_respiration: Optional[float] = None
    avg_stress_level: Optional[int] = None
    total_steps: Optional[int] = None
    has_checkin_today: bool = False

File: mobile/src/types/metrics.ts
export interface TodayMetrics {
  date: string
  readiness_score: number | null
  readiness_level: string | null
  hrv_last_night_avg: number | null
  hrv_status: string | null
  sleep_seconds: number | null
  avg_respiration: number | null
  avg_stress_level: number | null
  total_steps: number | null
  active_kcal: number | null
  total_kcal: number | null
  has_checkin_today: boolean
}
